design/ceph/object/bucketpolicy.md

```diff
@@ -0,0 +1,74 @@
+---
+title: adding bucket policy for ceph object store
+target-version: release-1.4
+---
+
+# Feature Name
+Adding bucket policy support for ceph object store
+
+## Summary
+The bucket policy is the feature in which permissions for specific user can be set on s3 bucket. Read more about it from [ceph documentation](https://docs.ceph.com/docs/master/radosgw/bucketpolicy/)
+
+Currently [ceph object store](/Documentation/ceph-object.md) can be consumed either via [OBC](/Documentation/ceph-object-bucket-claim.md) and [ceph object user](/Documentation/ceph-object-store-user-crd.md). As of now there is no direct way for ceph object user to access the OBC. The idea behind this feature to allow that functionality via Rook. Refer bucket policy examples from [here](https://docs.aws.amazon.com/AmazonS3/latest/dev/example-bucket-policies.html).  Please note it is different from `IAM policies`.
+
+
+## Proposal details
+
+Followings CRDs are needed to add for defining policies:
+  - _bucketPolicy_ in the `Spec` section of `CephObjectStoreUser`
+  - _bucketPolicy_ in the `parameters` section of `StorageClass` for `ObjectBucketClaim`
+
+Policies need to provide in generic `json` format. A policy can have multiple `statement`.
+
+Following checks are need to perform in Rook to verify whether the `bucketpolicy` applicable to OBC.
+- for ceph object user, `Principal` value should have username and `Resource` should have specific bucket names. It can be defined for buckets which are not part of the OBC as well, the `bucketname` of a OBC can be fetch from its `configmap`.
```

Comment:
```suggestion
- for ceph object user, `Principal` value should have a username and `Resource` should have specific bucket names. It can be defined for buckets which are not part of the OBC as well, the `bucketname` of an OBC can be fetched from its `configmap`.
```

Reply:
Done